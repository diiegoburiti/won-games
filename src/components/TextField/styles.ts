import styled, { css } from 'styled-components'
import { TextFieldProps } from '.'

type IconPositionProps = Pick<TextFieldProps, 'iconPosition'>

export const Wrapper = styled.section``

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    padding: 0 ${theme.spacings.xsmall};
    border: 0.2rem solid;
    border-radius: 0.2rem;
    border-color: ${theme.colors.lightGray};
    background: ${theme.colors.lightGray};

    &:focus-within {
      box-shadow: 0 0 0 0.5rem ${theme.colors.primary};
    }
  `}
`

export const Input = styled.input<IconPositionProps>`
  ${({ theme, iconPosition }) => css`
    width: 100%;
    color: ${theme.colors.black};
    font-family: ${theme.font.family};
    font-size: ${theme.font.sizes.medium};
    padding: ${theme.spacings.xxsmall} 0;
    padding-${iconPosition}: ${theme.spacings.xsmall};
    background-color: transparent;
    border: 0;
    outline: none;
  `}
`
export const Label = styled.label`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.black};
    cursor: pointer;
  `}
`

export const Icon = styled.div<IconPositionProps>`
  ${({ theme, iconPosition }) => css`
    display: flex;
    width: 2.2rem;
    color: ${theme.colors.gray};
    order: ${iconPosition === 'right' ? 1 : 0};

    & > svg {
      width: 100%;
      transition: 0.5s;

      &:hover {
        transform: rotate(360deg);
      }
    }
  `}
`
