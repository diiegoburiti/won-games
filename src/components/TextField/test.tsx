import { screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Email } from '@styled-icons/material-outlined'
import { renderWithTheme } from 'utils/tests/helpers'

import TextField from '.'

describe('<TextField />', () => {
  it('should render with label', () => {
    const { container } = renderWithTheme(
      <TextField label="E-mail" labelFor="email" />
    )

    expect(screen.getByRole('textbox')).toBeInTheDocument()
    expect(screen.getByText(/e-mail/i)).toHaveAttribute('for', 'email')
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render without label', () => {
    renderWithTheme(<TextField />)

    expect(screen.queryByLabelText('textbox')).not.toBeInTheDocument()
  })

  it('should render with placeholder', () => {
    const placehoolder = 'admin@gmail.com'

    renderWithTheme(<TextField placeholder={placehoolder} />)

    expect(screen.getByPlaceholderText(placehoolder)).toBeInTheDocument()
  })

  it('changes it`s values when typing', async () => {
    const onInput = jest.fn()
    renderWithTheme(
      <TextField
        onInput={onInput}
        label="TextField"
        labelFor="TextField"
        id="TextField"
      />
    )

    const input = screen.getByRole('textbox')
    const text = 'this is the new input value'
    userEvent.type(input, text)

    await waitFor(() => {
      expect(input).toHaveValue(text)
      expect(onInput).toHaveBeenCalledTimes(text.length)
    })
    expect(onInput).toHaveBeenCalledWith(text)
    //expect(screen.getByPlaceholderText(placehoolder)).toBeInTheDocument()
  })

  it('should be accessible by tab', () => {
    renderWithTheme(
      <TextField label="TextField" labelFor="TextField" id="TextField" />
    )

    const input = screen.getByLabelText('TextField')
    expect(document.body).toHaveFocus()

    userEvent.tab()
    expect(input).toHaveFocus()
  })

  it('should with icon', () => {
    renderWithTheme(<TextField icon={<Email data-testid="icon" />} />)

    expect(screen.getByTestId(/icon/i)).toBeInTheDocument()
  })

  it('should with icon on the right side', () => {
    renderWithTheme(
      <TextField iconPosition="right" icon={<Email data-testid="icon" />} />
    )

    expect(screen.getByTestId(/icon/i).parentElement).toHaveStyle({ order: 1 })
  })
})
