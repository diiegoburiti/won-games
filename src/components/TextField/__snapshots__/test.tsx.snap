// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render with label 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #EAEAEA;
  border-radius: 0.2rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c1:focus-within {
  box-shadow: 0 0 0 0.5rem #F231A5;
}

.c2 {
  width: 100%;
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.4rem;
  padding: 0.8rem;
  background-color: transparent;
  border: 0;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c0 {
  font-size: 1.4rem;
  color: #030517;
  cursor: pointer;
}

<section
  class=""
>
  <label
    class="c0"
    for="email"
  >
    E-mail
  </label>
  <div
    class="c1"
  >
    <input
      class="c2"
      type="text"
      value=""
    />
  </div>
</section>
`
